src/sdk.js

```diff
@@ -1,57 +1,99 @@
 window.Smartcar = (function(window) {
   'use strict';
+  // accesss denied error class
+  class AccessDenied extends Error { // eslint-disable-line no-restricted-syntax
+    constructor(message) {
+      super(message);
+      this.name = 'AccessDenied';
+    }
+  }
 
   /**
    * Initializes Smartcar class
    *
    * @param {Object} options - the SDK configuration object
    * @param {String} options.clientId - the applications' client id
-   * @param {String} options.redirectUri - the registered redirect uri of the application
+   * @param {String} options.redirectUri - the registered redirect uri of the
+   * application
    * @param {String[]} [options.scope] - requested permission scopes
-   * @param {Function} [options.onComplete] - called upon completion of the Auth flow
-   * @param {Boolean} [options.development=false] - launch Smartcar auth in development mode
-   * to enable the mock vehicle brand
+   * @param {Function} [options.onComplete] - called on completion of auth flow
+   * @param {Boolean} [options.development=false] - launch Smartcar auth in
+   * development mode to enable the mock vehicle brand
+   * @param {Boolean} [options.useSmartcarHostedRedirect=false] - use Smartcar's
```

Comment:
boolean flag to indicate whether their redirect is hosted by us. Any other/preferred naming conventions?

Reply:
I think this flag is descriptive. I don't know if there's anything I'd prefer to change it to.